### PR TITLE
Allow overriding the Referred-By header on REFER

### LINKED ISSUE #719: New Feature: Allow overriding Referred-By header
In our use case, a UA registers with a user ID, say `12345@host` and may receive calls on multiple DIDs. Our Asterisk ARI application delivers calls by dialling `12345@host` and setting the target DID in the `To` header.

When the call is answered and REFER'd later on, the `Referred-By` header is always `12345@host` (since that's how we are registered with the UA), but we need this to be the dialed DID@host. We need flexibility to override this header, its logic is currently hardcoded in #572.

### PR DESCRIPTION
**Pull request by jp1987**
Fixes #719

### CODE REVIEW

File: lib/RTCSession/ReferSubscriber.js [hunk 1]
```diff
@@ -54,10 +54,12 @@ module.exports = class ReferSubscriber extends EventEmitter
...
 
-    // Referred-By header field.
-    const referredBy = `Referred-By: <${this._session._ua._configuration.uri._scheme}:${this._session._ua._configuration.uri._user}@${this._session._ua._configuration.uri._host}>`;
-
-    extraHeaders.push(referredBy);
+    // Referred-By header field (if not already present).
+    if (!extraHeaders.some(header => header.startsWith('Referred-By:'))) 
+    {
+      const referredBy = `Referred-By: <${this._session._ua._configuration.uri._scheme}:${this._session._ua._configuration.uri._user}@${this._session._ua._configuration.uri._host}>`;
+      extraHeaders.push(referredBy);
```
Comment (jmillan) [inline lib/RTCSession/ReferSubscriber.js:61]:
```suggestion

      extraHeaders.push(referredBy);
```

**Comment by jmillan** [comment]
Thanks @jp1987!

**Comment by jmillan** [comment]
BTW, no new minor release has been created for this change.

I've just added this info in the current documentation: https://jssip.net/documentation/3.7.x/api/session/#method_refer

**Comment by jp1987** [comment]
> BTW, no new minor release has been created for this change.
> 
> I've just added this info in the current documentation: https://jssip.net/documentation/3.7.x/api/session/#method_refer

Thank you @jmillan!

**Comment by jp1987** [comment]
> BTW, no new minor release has been created for this change.
> 
> I've just added this info in the current documentation: https://jssip.net/documentation/3.7.x/api/session/#method_refer

You may want to change "Referred-by" to "Referred-By" in the documentation, it's case sensitive.

File: lib/RTCSession/ReferSubscriber.js [hunk 2]
```diff
@@ -54,10 +54,13 @@ module.exports = class ReferSubscriber extends EventEmitter
 
     extraHeaders.push(referTo);
 
-    // Referred-By header field.
-    const referredBy = `Referred-By: <${this._session._ua._configuration.uri._scheme}:${this._session._ua._configuration.uri._user}@${this._session._ua._configuration.uri._host}>`;
+    // Referred-By header field (if not already present).
+    if (!extraHeaders.some(header => header.startsWith('Referred-By:')))
```
Comment (ibc) [inline lib/RTCSession/ReferSubscriber.js:58]: This is wrong. It should not be case sensitive. SIP headers are not case sensitive.

Comment (jp1987) [reply lib/RTCSession/ReferSubscriber.js:58]: Then it should be `header.toLowerCase().startsWith('referred-by:')`. New PR or can you guys fix this in master?

Comment (jmillan) [reply lib/RTCSession/ReferSubscriber.js:58]: > SIP headers are not case sensitive

Absolutely right.

Fixed in new release.

**Pull request merged**